services/cd-service/pkg/service/version.go

```diff
@@ -72,3 +75,58 @@ func (o *VersionServiceServer) GetVersion(
 	}
 	return &res, nil
 }
+
+func (o *VersionServiceServer) GetManifests(ctx context.Context, req *api.GetManifestsRequest) (*api.GetManifestsResponse, error) {
+	if req.Application == "" {
+		return nil, status.Error(codes.InvalidArgument, "no application specified")
+	}
+
+	state := o.Repository.State()
+
+	wrapError := func(what string, err error) error {
+		if !os.IsNotExist(err) {
+			return status.Errorf(codes.NotFound, "%s not found", what)
+		} else {
+			return status.Error(codes.Internal, err.Error())
+		}
+	}
+
+	var release uint64
+	if req.Release == "latest" {
+		allReleases, err := state.GetApplicationReleases(req.Application)
+		if err != nil {
+			return nil, wrapError("application", err)
+		}
+
+		if len(allReleases) == 0 {
+			return nil, status.Errorf(codes.NotFound, "no releases found for application %s", req.Application)
+		}
+
+		for _, r := range allReleases {
+			if r > release {
+				release = r
+			}
+		}
```

Comment:
this looks like it can be simplified by calling `GetLastRelease`

Reply:
Did not see that function :see_no_evil:

Reply:
Is it okay to call it like `GetLastRelease(state.Filesystem, applicationName)`? It feels wrong to access `.Filesystem` directly while in other cases we simply call a function on `*State`. It seems inconsistent. How should we proceed?